File: Breaking the Records/breakingRecords.cpp
#include <iostream>
#include <vector>
#include <algorithm>

std::vector<int> breakingRecords(const std::vector<int>& scores) {
    int minimum = scores[0];
    int maximum = scores[0];
    std::vector<int> record(2, 0); // [max_breaks, min_breaks]

    for (size_t i = 1; i < scores.size(); ++i) {
        if (scores[i] > maximum) {
            maximum = scores[i];
            record[0]++;
        }
        if (scores[i] < minimum) {
            minimum = scores[i];
            record[1]++;
        }
    }

    return record;
}

int main() {
    std::vector<int> scores = {12, 24, 10, 24};
    std::vector<int> result = breakingRecords(scores);
    std::cout << "Max Breaks: " << result[0] << ", Min Breaks: " << result[1] << std::endl;
    return 0;
}

File: Breaking the Records/breakingRecords.py
def breakingRecords(scores):
    minimum = scores[0]
    maximum = scores[0]
    record = [0, 0] # max_breaks, min_breaks
    
    for score in scores[1:]:
        if score > maximum:
            maximum = score
            record[0] += 1
        if score < minimum:
            minimum = score
            record[1] += 1
    return record


scores = [12,24,10,24]
print(breakingRecords(scores))
